ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -222,10 +237,11 @@ runWith :: forall m addrNTN addrNTC versionDataNTN versionDataNTC blk.
      , IOLike m, MonadTime m, MonadTimer m
      , Hashable addrNTN, Ord addrNTN, Typeable addrNTN
      )
-  => RunNodeArgs m addrNTN addrNTC blk
+  => NetworkP2PMode
+  -> RunNodeArgs m addrNTN addrNTC blk
   -> LowLevelRunNodeArgs m addrNTN addrNTC versionDataNTN versionDataNTC blk
-  -> m Void
-runWith RunNodeArgs{..} LowLevelRunNodeArgs{..} =
+  -> m ()
+runWith enableP2P RunNodeArgs{..} LowLevelRunNodeArgs{..} =
```

Comment:
Please add `enableP2P` as a field in `RunNodeArgs` instead of adding it as a new separate parameter to `run` and `runWith`.

Reply:
As a field, you can give it a polarized name, so it can just be a `Bool` if you wish. eg `rnEnableP2P :: Bool`.

Reply:
I think I prefer having the `NetworkP2PMode` type, if that is okay